src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellPageRendererTracker.cs

```diff
@@ -134,8 +134,7 @@ protected virtual void OnPagePropertyChanged(object sender, PropertyChangedEvent
 
 		protected virtual void UpdateTabBarVisible()
 		{
-			var tabBarVisible =
-				(Page.FindParentOfType<ShellItem>() as IShellItemController)?.ShowTabs ?? Shell.GetTabBarIsVisible(Page);
+			var tabBarVisible = Shell.GetTabBarIsVisible(Page) ? true : (Page.FindParentOfType<ShellItem>() as IShellItemController)?.ShowTabs ?? false;
```

Comment:
Comparing this to Android. It looks like this works on android because we are using the DisplayedPage observer vs subscribing to "appearing" 

https://github.com/dotnet/maui/blob/7f292e66c8b471870d14099c6b4d1058c4f63b5a/src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellItemRendererBase.cs#L60

If we modify `ShellPageRendererTracker` so it calls `UpdateShellToMyPage();` via `((IShellSectionController)ShellSection).AddDisplayedPageObserver(this, UpdateDisplayedPage);` will that work? or does that all happen too late and won't visually look good?

Reply:
@PureWeen ,No, it will not. Upon analysis, we have determined that both Android and iOS function similarly. The
AddDisplayedPageObserver is added to the page during the ShellSectionChanged event, and iOS contains comparable code. However, it does not trigger notifications when navigating between pages.

Reply:
Are you saying that Android does trigger those notifications but iOS doesn't?

Reply:
@PureWeen, Apologies for the confusion earlier. In our case, we have a series of pages that navigate using `GoToAsync`. On both Android and iOS platforms, we've added the `AddDisplayPageObserver` during section changes for the initial load. For subsequent navigations, we receive notifications from the `GoToAsync` method.
 
 The fix addresses the `UpdateTabBarIsVisible` behavior on iOS, considering both TabBarIsVisible and ShowTabs, which aligns with the `UpdateTabBarVisibility` handling on Android.